sotodlib/site_pipeline/update_smurf_caldbs.py

```diff
@@ -109,15 +123,252 @@ def smurf_detset_info(config:str | dict,
             })
         write_dataset(
             det_rs, 
-            config['archive']['h5file'], 
+            config['archive']['detset']['h5file'], 
             ts.name, 
             overwrite,
         )
         # add new entries to database
         if ts.name not in added_detsets:
             db_data = {'dets:detset': ts.name,
                     'dataset': ts.name}
-            db.add_entry(db_data, config['archive']['h5file'])
+            db.add_entry(db_data, config['archive']['detset']['h5file'])
+
+
+def get_cal_obsids(ctx, obs_id, cal_type):
+    """
+    Returns set of obs-ids corresponding to the most recent calibration
+    operations for a given obsid:
+
+    Returns
+    ----------
+        obs_ids: dict
+            Dict of obs_ids for each detset in specified operation
+    """
+    obs = ctx.obsdb.query(f"obs_id == '{obs_id}'")[0]
+    detsets = ctx.obsfiledb.get_detsets(obs_id)
+    min_ct = obs['start_time'] - 3600*24*7
+    cal_all = ctx.obsdb.query(
+        f"""
+        start_time <= {obs['start_time']} and subtype=='{cal_type}'
+        and start_time > {min_ct}
+        """, sort=['start_time']
+    )[::-1]
+
+    obs_ids = {
+        ds: None for ds in detsets
+    }
+
+    for o in cal_all:
+        dsets = ctx.obsfiledb.get_files(o['obs_id']).keys()
+        for ds in dsets:
+            if ds in obs_ids:
+                if obs_ids[ds] is None:
+                    obs_ids[ds] = o['obs_id']
+
+    return obs_ids
+
+# Dtype for calibration set
+cal_dtype = [
+    ('dets:readout_id', '<U40'),
+    ('dets:cal.r_tes', float),
+    ('dets:cal.r_frac', float),
+    ('dets:cal.p_bias', float),
+    ('dets:cal.s_i', float),
+    ('dets:cal.bg', int),
+    ('dets:cal.bg_polarity', int),
+    ('dets:cal.r_n', float),
+    ('dets:cal.p_sat', float),
+]
+
+@dataclass
+class CalInfo:
+    # Fields must be ordered like cal_dtype!
+    readout_id: str = ''
+
+    # From bias steps
+    r_tes: float = np.nan   # Ohm
+    r_frac: float = np.nan
+    p_bias: float = np.nan  # J
+    s_i: float = np.nan     # 1/V
```

Comment:
Why do you get `s_i` only from the IV? We know the current IV + BS analysis differ and I think we believe that the BS calibration is more valid at least for rfrac ~< 0.6. Can we include both? Maybe `s_i_iv` and `s_i_bs`